_upgrading/sopel-8.md

```diff
@@ -0,0 +1,428 @@
+
+## Owner/admin usage changes
+
+### Updated Python requirements & support policy
+
+Sopel 8.0 requires Python 3.8 or higher. Support for Python 2.7 and 3.3–3.7 has
+been removed. In exchange, known issues sometimes preventing use of Sopel 7 with
+Python 3.11+ have been fixed.
+
+During the lifecycle of Sopel 8.x, we will add new Python releases to our test
+suite as soon as possible. We will keep testing against end-of-life Python
+versions unless and until it becomes technically impossible to do so. However,
+if testing against an EOL Python version does become impossible, we will drop it
+in the next maintenance release of Sopel.
+
+### CLI changes
+
+Sopel 8 continues the command-line interface overhaul we began in version 7,
+mostly in the form of removing support for legacy usage from Sopel's 6.x era.
+
+The bare `sopel` command now offers only basic control. Most of its legacy
+options have been removed:
+
+|             Legacy command            |             Modern command             |
+|:-------------------------------------:|:--------------------------------------:|
+| `sopel --quit` or `sopel -q`          | `sopel stop`                           |
+| `sopel --kill` or `sopel -k`          | `sopel stop --kill` or `sopel stop -k` |
+| `sopel --restart` or `sopel -r`       | `sopel restart`                        |
+| `sopel --configure-all` or `sopel -w` | `sopel configure`                      |
+| `sopel --configure-modules`           | `sopel configure --plugins`            |
+| `sopel --list` or `sopel -l`          | `sopel-config list`                    |
+| `sopel -v`                            | `sopel -V` / `sopel --version`         |
+| `sopel --quiet`                       | n/a; this feature didn't work anyway   |
+
+### Ignore system
+
+In Sopel 8, "hostmask" blocks are now called "host" blocks, which reflects the
+reality of how those values are used by Sopel. Bot admins' muscle memories will
+have to learn to type e.g. `.blocks add host` instead of `.blocks add hostmask`.
+
+We intend to add [actual "hostmask" blocking][better-ignore-system] in a future
+Sopel release.
+
+_Note that the **config file** option for "hostmask" blocks already was, and
+still is, named `host_blocks`. This change only affects interactive blocklist
+editing by bot admins via IRC commands._
+
+  [better-ignore-system]: https://github.com/sopel-irc/sopel/issues/1355
+
+### Configuration & plugin-handling changes
+
+#### What's a "module", doc?
+
+Sopel 8 no longer loads plugins from the `<homedir>/modules` directory by default.
+To migrate, rename this directory to `plugins/` or declare it in the `extra` field of the
+core configuration section.
+
+This is part of our efforts to clarify the difference between a "module" (which is a
+[Python concept][py-module]) and a "plugin" (which is a special kind of "module" that
+can extend Sopel with new functionality).
+
+  [py-module]: https://docs.python.org/3/glossary.html#term-module
```

Comment:
The new `core-section` link doesn't exist yet on the live site, but the page was moved for 8.0 in sopel-irc/sopel#2533.

```suggestion
Sopel 8 no longer loads plugins from the `<homedir>/modules` directory by default.
To migrate, rename this directory to `plugins/` or declare it in the `extra` field of
[the [core] configuration section][core-section].

This is part of our efforts to clarify the difference between a "module" (which is a
[Python concept][py-module]) and a "plugin" (which is a special kind of "module" that
can extend Sopel with new functionality).

  [core-section]: https://sopel.chat/docs/run/configuration.html
  [py-module]: https://docs.python.org/3/glossary.html#term-module
```